test/specs/cutting.js

```diff
@@ -1,30 +1,55 @@
-import {load, finputReversedDelimiters, finputDefaultDelimiters} from '../pageObjects/index';
-import customCommandsFactory from '../customCommands';
+import { load, unload } from '../pageObjects/index';
+import { getDriver } from '../helpers';
+import { cutting, itCutting } from '../customCommands';
 
 describe('cutting', () => {
-  beforeAll(load);
+
+  let driver;
+  let finputDefaultDelimiters;
+  let finputReversedDelimiters;
+
+  beforeAll(async () => {
+    driver = getDriver();
+    ({ 
+      finputDefaultDelimiters, 
+      finputReversedDelimiters 
+    } = await load(driver));
+  });
+
+  afterAll(async () => await unload(driver));
 
   describe('default delimiters', () => {
-    const {cutting} = customCommandsFactory(finputDefaultDelimiters);
+
+    beforeAll(() => {
+      itCutting.expectCutting = cutting({
+        driver,
+        finputElement: finputDefaultDelimiters
+      });
+    });
 
     // Cutting from input (should fully format unless no characters selected)
     // None selected
-    cutting(0).characters().from(`123456`).startingFrom(0).shouldShow(`123,456`);
-    cutting(2).characters().from(`12`).startingFrom(2).shouldShow(`12`);
-
-    cutting(4).characters().from(`123456`).startingFrom(1).shouldShow(`156.00`);
-    cutting(5).characters().from(`1234`).startingFrom(0).shouldShow(``);
+    itCutting({ cut: 0, startingFrom: 0, tested: `123456`, expected: `123,456` });
```

Comment:
This refactor means we lose the bdd style syntax. The prior read as "cutting zero characters from '123456' starting from 0 should show '123,456'" and is very easy to understand the intent of the test. Whereas the new syntax requires knowing about the itCutting function and how the params are used. Is it not possible to keep the bdd type syntax by having the driver passed as a param to the customCommandsFactory? Perhaps I'm missing some other consideration that means this isn't possible.

Reply:
I do agree the previous syntax was better, although it was coupling the `it` call to the function passed to the `it`. And you still needed to know about cutting, characters, from, startingFrom, shouldShow, etc... Also, the main and perhaps real disadvantage was that the cutting wrapper needed to be defined statically.
The point of this commit is to have an expectation function separate from the `it` call. The expectation function is defined in the `beforeAll`. the `itCutting` is cumbersome, I agree, and I can remove it and add the text explicitely here, but at least it has the advantage to provide easy composition. Also, the api about the scenario under test is quite explicit.
Any thoughts ?

Reply:
It is my understanding that we want to be able to inject `driver` into each spec while being able to `load` before all specs and `unload` after all specs. I see your point about decoupling `expect` from `it` and how the proposed change uses that to inject the `driver` into the `cutting` function meaning each spec can use a different `driver` in the future.

I was wondering though if its possible to achieve this without changing the bdd style syntax. If we only want to be able to set the driver per `description` block then can `driver` just be injected as a param into the pre-existing `customCommandsFactory`? If we want to be able to set `driver` at the per spec level then would it be possible to add something like a `usingDriver` function to `chainFunctions` within the custom command that sets the `driver` for that spec (giving something like `cutting(...).characters().usingDriver(...)...`)? Perhaps I'm missing some consideration that means this wouldn't work though?

Reply:
ok let me try to come up with something, I will open another PR